### DISCUSSION TITLE
Is the storage concurrency-safe?

### DISCUSSION THREAD

**Discussion by silverwind** [verdaccio/verdaccio#3907] [Q&A] [answered]
Is it possible for multiple verdaccio application to safely share disk-based storage? Specifically, I'm looking to run multiple verdaccio docker containers which share a bind-mounted  `/verdaccio/storage` volume through the docker host.

**Comment by juanpicado** [comment, marked as answer]
Nop, the storage files are locked while are written, thus disk based with multiple instances is not supported.

**Comment by silverwind** [comment]
Thanks! Could it work when a load balancer split reads from writes based on HTTP method and then:

1. Direct writes to the same instance
2. Direct reads to all instances

This is assuming GET and HEAD do not lock, is this a correct assumption?

**Reply by silverwind** [reply]
I assume the answer is no because published packages would only go to one instance.

Really looking for ways to scale verdaccio because with its single-threaded nature, it bottlnecks on CPU very quickly with just 3-4 npm clients bombarding it (especially npm v10 is bad at this, see https://github.com/npm/cli/issues/7072).
